core/src/main/java/org/infinispan/health/impl/CacheHealthImpl.java

```diff
@@ -9,36 +8,36 @@
 
 public class CacheHealthImpl implements CacheHealth {
 
-    private final AdvancedCache<?, ?> cache;
+   private final Cache<?, ?> cache;
 
-    public CacheHealthImpl(Cache<?, ?> cache) {
-        this.cache = cache.getAdvancedCache();
-    }
+   public CacheHealthImpl(Cache<?, ?> cache) {
+      this.cache = cache;
+   }
 
-    @Override
-    public String getCacheName() {
-        return cache.getName();
-    }
+   @Override
+   public String getCacheName() {
+      return cache.getName();
+   }
 
-    @Override
-    public HealthStatus getStatus() {
-        if (!isComponentHealthy() || cache.getAvailability() == AvailabilityMode.DEGRADED_MODE) {
-            return HealthStatus.UNHEALTHY;
-        }
-        DistributionManager distributionManager = SecurityActions.getDistributionManager(cache);
-        if (distributionManager != null && distributionManager.isRehashInProgress()) {
-            return HealthStatus.REBALANCING;
-        }
-        return HealthStatus.HEALTHY;
-    }
+   @Override
+   public HealthStatus getStatus() {
+      if (!isComponentHealthy() || cache.getAdvancedCache().getAvailability() == AvailabilityMode.DEGRADED_MODE) {
```

Comment:
Why do we need `cache.getAdvancedCache()` here?

Reply:
this is something I wanted to discuss on the PR.
the CacheHealthImpl has originally advancedCache instead of cache
When I looked to the code of `SecurityActions.getDistributionManager(cache);`

the code does again `getAdvancedCache`
so we where doing `getAdvancedCache` and again `getAdvancedCache`

I used mockito for the rebalancing cover. That's how I realised.